hr_timesheet_sheet/models/hr_timesheet_sheet.py

```diff
@@ -353,14 +382,6 @@ def write(self, vals):
                 rec.delete_empty_lines(True)
         return res
 
-    @api.multi
-    def name_get(self):
```

Comment:
Why not reuse the name_get ()? Can't the display_name be used instead of adding another new field?

Reply:
@astirpe due to `@api.depends('date_start', 'date_end')`?

Reply:
`name_get` is a special method that is called always, so no need of depends. This, however, can benefit from caching techniques.

Reply:
what about https://github.com/odoo/odoo/blob/71faa19af06d5002be627adbfd92dc25941c69c6/odoo/models.py#L1569 ?

Reply:
It's get cached into a field, no? that follows same rules?

Reply:
Not exactly. Logic is diffuse here and I admit this confuses me also, but you can try and experiment. My experience in this is that you can put whatever in `name_get` and works properly. Example:

https://github.com/OCA/l10n-spain/blob/12.0/l10n_es_partner/models/res_partner.py#L62

Reply:
And why do you need an @api.depends if the calculated field is not stored?

Reply:
@astirpe this is for cache invalidation. Although the value is not stored, you need to ensure when this is going to be recomputed. If in all the chain there's no api.depends, it will be computed always, but it's a good habit to have a depends for reducing scope or avoiding side effects.

Reply:
TIL
But there's no way to do the same on a name_get () ?

Reply:
I have seen sometimes in some computed methods the use of an empty `@api.depends()` to not break that chain.

Reply:
> But there's no way to do the same on a name_get () ?

I haven't dug enough in this part to say.

Reply:
At least that's not the documented way?

Reply:
Looking at the code of standard Odoo and on OCA modules, I can see that the `name_get()` method is sometimes defined with `@api.depends` decorator and nobody complained about that so far. So I try again: is it possible to reuse the `name_get()` combined with the `@api.depends` decorator and make use of the `display_name` field, without the need of adding a new extra field? I already asked this question but I didn't get a final answer, and I just saw the "Marked as resolved" to my question.

Reply:
I haven't seen any proof why that would work in the `odoo/models.py` or in the documentation, thus this `@api.depends` decorator seems to be just ignored, as `name_get()` is not mentioned as a compute-method anywhere. We could override `_compute_display_name` directly, yet that would imply that `name_get()` (that other modules may call directly) would return incorrect result.

Reply:
So as a conclusion, from my understanding, having extra field is the safest way possible, and from the documentation - only possible one?

Reply:
As said, if you only defined `name_get`, you will get the same result displaying that when on any many2one widget, but without caching. That depends is for the `display_name` field, which is another thing.

Reply:
By setting a `_rec_name`, now the existing field "`name`" becomes unused. So field `name` could be removed now. Or even better, can be reused instead of adding a new field `sheet_name`. At least can this field `name` be reused?

Reply:
I don't think `name` was used properly - as it's actually a "note" as far as I can see from the code. So it should've been `note` all along

Reply:
Looking at the previous versions of the module, it seems to me that the field `name` was supposed to be just the name field. Probably the `load='_classic_write'` that is present in the `name_get()` was part of some old mechanism used to write on that field. Nobody is using any field for the notes because the chatter is there for that purpose.

Reply:
This is the only reference I've found: https://github.com/OCA/hr-timesheet/blob/e4f0653d8596a004bbb94c6090ddaf0c5fe43006/hr_timesheet_sheet/views/hr_timesheet_sheet_views.xml#L47

if so, I may actually use the `name` field?

Reply:
Yes, I prefer this way :+1:  Give me some time to test it.

Reply:
Works fine. :+1: 

This line can be removed:
https://github.com/OCA/hr-timesheet/blob/e4f0653d8596a004bbb94c6090ddaf0c5fe43006/hr_timesheet_sheet/views/hr_timesheet_sheet_views.xml#L47

Reply:
Done!

Reply:
Oh and ... you can also remove `string='Name',`